framework/src/utils/MooseEnum.C

```diff
@@ -46,62 +43,56 @@ MooseEnum::withNamesFrom(const MooseEnumBase & other_enum)
 /**
  * Private constuctor for use by libmesh::Parameters
  */
-MooseEnum::MooseEnum() : _current_id(INVALID_ID) {}
+MooseEnum::MooseEnum() : _current("", INVALID_ID) {}
 
-MooseEnum::MooseEnum(const MooseEnumBase & other_enum) : MooseEnumBase(other_enum) {}
+MooseEnum::MooseEnum(const MooseEnumBase & other_enum)
+  : MooseEnumBase(other_enum), _current("", INVALID_ID)
```

Comment:
@aeslaughter - Thanks for the other fixups. I'm still puzzled by this one. If we copy construct a MooseEnum from another (that's valid), the new one isn't valid? Is that intentional? Why?